### PR TITLE
Improve music trivia matching

### PR DESCRIPTION
**Pull request by rafaeldamasceno**
Changes the matching between the user input and the content, normalizing all strings to be checked:

- Changes the strict equal to be more lenient using includes. For example if the current song is "I Ran (So Far Away)", but the definition in the JSON is only "I Ran" and the user types "I Ran (So Far Away)", the check will pass because the result is included in the user guess.
- The first normalization step removes diacritics. "Céline Dion" becomes "Celine Dion".
- After diacritics have been removed, we can bring all the letters to lowercase.
- The second normalization step removes non alphanumeric characters. Helps matching strings with ampersands, apostrophes, dashes, etc. For example, "AC/DC" becomes "ACDC".
- Trims the strings to remove extra whitespace at the start and end of the strings.
- Removes double whitespace in the middle of the strings ("The  Beatles" becomes "The Beatles").
- Now that the string is all tidied up, we can finally bring the characters to all lowercase for the comparison.

### CODE REVIEW

**Comment by Bacon-Fixation** [comment]
great job <3

now i can stand a chance to win in music trivia :D

Much Love
-Bacon

File: commands/music/music-trivia.js [hunk 1]
```diff
@@ -200,20 +208,13 @@ module.exports = class MusicTriviaCommand extends Command {
...
-              msg.content.toLowerCase() ===
-                queue[0].title.toLowerCase() +
-                  ' ' +
-                  queue[0].singer.toLowerCase()
-            ) {
-              if (
-                (songSingerFound && !songNameFound) ||
-                (songNameFound && !songSingerFound)
-              ) {
+            }
```
Comment (galnir) [inline commands/music/music-trivia.js:211]: Can you explain lines 212-217? Line 212 says 'if user guessed song name' but line 217 checks for both name and singer

File: commands/music/music-trivia.js [hunk 2]
```diff
@@ -173,13 +180,14 @@ module.exports = class MusicTriviaCommand extends Command {
               message.guild.triviaData.triviaScore.set(
                 msg.author.username,
                 message.guild.triviaData.triviaScore.get(msg.author.username) +
-                  1
+                  2
```
Comment (galnir) [inline commands/music/music-trivia.js:183]: Why did you change the number from 1 to 2 here?

File: commands/music/music-trivia.js [hunk 3]
```diff
@@ -155,12 +155,19 @@ module.exports = class MusicTriviaCommand extends Command {
...
-              if (songNameFound && songSingerFound) {
+            let guess = normalizeValue(msg.content);
+            let title = normalizeValue(queue[0].title);
+            let singer = normalizeValue(queue[0].singer);
+            
+            // if user guessed both
+            if (
+              guess.includes(singer) && guess.includes(title)
+            ) {
+              if (
```
Comment (galnir) [inline commands/music/music-trivia.js:167]: The if statement from line 164 checks if the user guessed both singer and title but the if statement from line 168 (which is inside the block the 'if' from 164 created) checks for one of them?

**Review by galnir — COMMENTED** [review]
Can you clarify some things? Take a look at my comments

File: commands/music/music-trivia.js (hunk 3 above)
Comment (rafaeldamasceno) [reply commands/music/music-trivia.js:167]: The if statement on 164 is checking if the user's guess contains both the singer and the title. The if statement on 167 is checking if any of these has previously been guessed, so that if someone had already guessed any of those, only 1 point is attributed instead of the two.

This block is unchanged from the original code, I just moved the if clause that checks both things to the beginning and the new condition is based on .contains() and not a strict comparison. Otherwise, any of the singer/title checks would always pass the condition before the check for both.

File: commands/music/music-trivia.js (hunk 2 above)
Comment (rafaeldamasceno) [reply commands/music/music-trivia.js:183]: Again, I moved the entire block to the top :) It's just that git does very nice diffs. But the number changed because the test for both things is now in this first if statement, so if no one had guessed anything and you instantly get both, you get the two points.

File: commands/music/music-trivia.js (hunk 1 above)
Comment (rafaeldamasceno) [reply commands/music/music-trivia.js:211]: This is the same logic as before. The if clause on 213 checks if the _user's guess_ is correct for the song name. The if clause on 214 check's if the song name had already been previously guessed or not. The if clause on 217 tests to see if both things have been guessed now and if it should move into the next song.

**Comment by rafaeldamasceno** [comment]
> Can you clarify some things? Take a look at my comments

I think I replied to all of your comments. As a more general clarification, I haven't really changed of the game's code or logic besides the string normalization and then the if conditions from === to .contains(). It's just that I had to reorder the if clauses to test if the user guessed both things to the top, and now Git is producing diffs that seem to make no sense :) In reality, the code is the same and it still looks correcto to me.

**Review by galnir — APPROVED** [review]
Thanks for your contribution @rafaeldamasceno 💯
